Notes.md

```diff
@@ -0,0 +1,50 @@
+##  Questions
+
+1. In CoreDataStorage.swift, what would be a better approax to not have such a convoluted file dealing with ToDos (insert, delete, update...), Categories and future Entities?
+
+
+2. On AddTodoView I have an state that is when the item is saved. On the view at first I wrote this
```

Comment:
In general state management surrounding dismissing views feels a little iffy to me in SwiftUI. So far my favorite way has been to pass a binding to the property that determines whether a sheet is presented to the sheet. When the sheet sets that bound property to `false`, its parent will dismiss. Slightly neater than using `wrappedValue.dismiss()` in my opinion.

Otherwise, I think both of your approaches look alright and it depends on what you want to do exactly. Not sure whether the first actually works though, it looks like the user wouldn't see the `Text` before your sheet is dismissed. If that's the case I think the `onReceive` approach you showed is a but cleaner. (with the delay the first would work too, I think that would be kind of neat for informing the user before dismissing like you mentioned)